blueprints/codes/eurocode/nen_en_1992_1_1_c2_2011/chapter_5_structural_analysis/formula_5_1.py

```diff
@@ -0,0 +1,130 @@
+"""Formula 5.1 from NEN-EN 1992-1-1+C2:2011: Chapter Structural Analysis."""
+# pylint: disable=arguments-differ
+import numpy as np
+
+from blueprints.codes.eurocode.nen_en_1992_1_1_c2_2011 import NEN_EN_1992_1_1_C2_2011
+from blueprints.codes.formula import Formula
+from blueprints.type_alias import M
+
+
+class Form5Dot1Imperfections(Formula):
+    """Class representing formula 5.1 for the calculation of initial inclination imperfections, Θi."""
+
+    label = "5.1"
+    source_document = NEN_EN_1992_1_1_C2_2011
+
+    def __init__(
+        self,
+        theta_0: float,
+        alpha_h: float,
+        alpha_m: float,
+    ) -> None:
+        """[Θi] Initial inclination imperfections, Θi, is a ratio between height and inclination of the member [-].
+
+        NEN-EN 1992-1-1+C2:2011 art.5.2(5) - Formula (5.1)
+
+        Parameters
+        ----------
+        theta_0 : float
+            [Θ0] Basic value [-].
+            Note: The value of Θ0 for use in a Country may be found in its National Annex.
+            The recommended value is 1/200
+        alpha_h : float
+            [αh] Reduction factor for length or height [-].
+            Use your own implementation of this value or use the SubForm5Dot1ReductionFactorLengthOrHeight class.
+        alpha_m : float
+            [αm] Reduction factor for number of members [-].
+            Use your own implementation of this value or use the SubForm5Dot1ReductionFactorNumberOfMembers class.
+        """
+        super().__init__()
+        self.theta_0 = theta_0
+        self.alpha_h = alpha_h
+        self.alpha_m = alpha_m
+
+    @staticmethod
+    def _evaluate(
+        theta_0: float,
+        alpha_h: float,
+        alpha_m: float,
+    ) -> float:
+        """Evaluates the formula, for more information see the __init__ method"""
+        if theta_0 < 0:
+            raise ValueError(f"Negative theta_0: {theta_0}. theta_0 cannot be negative")
+        if alpha_h < 0:
+            raise ValueError(f"Negative alpha_h: {alpha_h}. alpha_h cannot be negative")
+        if alpha_m < 0:
```

Comment:
replace with generic validation methods

Reply:
I will when the branch with validations is merged.

Reply:
so not resolved yet: unresolved conv